sound/soc/sof/sof-of-dev.c

```diff
@@ -77,7 +81,13 @@ int sof_of_probe(struct platform_device *pdev)
 		sof_pdata->fw_filename = desc->default_fw_filename[SOF_IPC];
 	}
 
-	sof_pdata->tplg_filename = desc->default_tplg_filename[SOF_IPC];
+	if (tplg_filename) {
+		sof_pdata->tplg_filename = tplg_filename;
+		dev_dbg(dev, "Module parameter used, change tplg filename to %s\n",
+			sof_pdata->tplg_filename);
+	} else {
+		sof_pdata->tplg_filename = desc->default_tplg_filename[SOF_IPC];
+	}
```

Comment:
I would actually not do this. you can follow the example of the nocodec topology, and use DT/OF to set the topology name UNLESS it's been overridden by the user.

In other words, you can keep your existing DT/OF mechanism and just provide a means to override it. You do not need to define and hard-code a default topology name.

Reply:
@plbossart the problem with our existing DT/OF mechanism was that it was not accepted by Mark so it is only in our internal patches.

https://www.spinics.net/lists/devicetree/msg433913.html

Reply:
maybe, but hard-coding a default tplg name isn't right either. Why would the wm8960 be a good choice?

````
.default_tplg_filename = {
￼		[SOF_IPC] = "sof-imx8mp-wm8960.tplg",
￼	},
````

Reply:
Yes, but this is hardoced for the current machine. And in NXP speaking this means the Evaluation Kit. It usually comes like this:

- CPU Board (SoC + basic peripherals, eg. wm8960) + Base Board (additional peripherals, here we have cs42888).

Vendor will use the SoC and might decide to change the peripherals. But at that moment we are talking about a different machine so they will need to create separate DTBs and separate ```sof_dev_desc``` structures.

Reply:
My point is that sof_dev_desc was supposed to be a descriptor of the SOC and hence is part of the platform driver. The topology needs to match what the machine driver exposes. Different layers.

I have no real objection if this is what you want to do, I just don't see how Mark is going to like it better.

Reply:
I see. I will try to find a way to expose this in the machine driver then. We use the generic simple machine driver. Will need open a discussion with Mark for this, then.

@paulstelian97 just  adapt this patch to only override tplg filename is param is present and we will find an alternate way to specify the default topology.